File: gpt-3.5-ft-small/context/enter_data_directive.c
#ifndef T1
//T1:enter_data,data,data-region,V:1.0-2.7
int test1(){
    int err = 0;
    int device_num = acc_get_device_num();
    int* a = (int*)malloc(10 * sizeof(int));
    real_t * b = (real_t *)malloc(10 * sizeof(real_t));

    for (int x = 0; x < 10; ++x){
        a[x] = 1;
        b[x] = 1.0;
    }

    #pragma acc enter data copyin(a[0:10], b[0:10])

    #pragma acc data present(a[0:10], b[0:10])
    {
        #pragma acc parallel present(a[0:10], b[0:10])
        {
            #pragma acc loop
            for (int x = 0; x < 10; ++x){
                a[x] = 5;
            }
            #pragma acc loop
            for (int x = 0; x < 10; ++x){
                b[x] = 5.0;
            }
        }
    }
    #pragma acc exit data copyout(a[0:10], b[0:10])

    for (int x = 0; x < 10; ++x){
        if (a[x] != 5){
            err += 1;
        }
        if (fabs(b[x] - 5.0) > PRECISION){
            err += 1;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/template_context_full/enter_data_directive.f90
PROGRAM main
      IMPLICIT NONE
      INTEGER :: failcode, testrun
      LOGICAL :: failed
      INCLUDE "acc_testsuite.Fh"
      REAL(8),DIMENSION(10) :: a, b, c
      failed = .FALSE.
      failcode = 0
#ifndef T1
      LOGICAL :: test1
#endif
      INTEGER :: dev_present

      dev_present = 1
      !$acc enter data copyin(a(1:10), b(1:10), c(1:10))
      !$acc parallel present(a(1:10), b(1:10), c(1:10))
        !$acc loop
        DO testrun = 0, 9
          a(testrun + 1) = 2.0
        END DO
        !$acc loop
        DO testrun = 0, 9
          b(testrun + 1) = 2.0
        END DO
        !$acc loop
        DO testrun = 0, 9
          c(testrun + 1) = 2.0
        END DO
      !$acc end parallel

      DO testrun = 0, 9
        IF (abs(a(testrun + 1) - 2.0) .gt. PRECISION) THEN
          dev_present = 0
        END IF
        IF (abs(b(testrun + 1) - 2.0) .gt. PRECISION) THEN
          dev_present = 0
        END IF
        IF (abs(c(testrun + 1) - 2.0) .gt. PRECISION) THEN
          dev_present = 0
        END IF
      END DO

      IF (dev_present .eq. 0) THEN
        failcode = failcode + 2**1
      END IF
#ifndef T1
      DO testrun = 1, NUM_TEST_CALLS
        failed = failed .or. test1
      END DO
#endif
      EXIT (failcode)
      END PROGRAM


      LOGICAL FUNCTION test1()
        IMPLICIT NONE
        INTEGER :: errors
        INCLUDE "acc_testsuite.Fh"
        INTEGER :: testrun, dev_present
        REAL(8),DIMENSION(10) :: a, b, c
        dev_present = 1
        errors = 0

        DO testrun = 0, 9
          a(testrun + 1) = 1
          b(testrun + 1) = 1
          c(testrun + 1) = 1
        END DO

        $OMP_TARGET_ENTER_DATA MAPTO(a(0:9),b(0:9),c(0:9))
        $OMP_TARGET
        $OMP_PARALLEL
          $OMP_PARALLEL FOR
          DO testrun = 0, 9
            a(testrun + 1) = 2
          END DO
          $OMP_PARALLEL FOR
          DO testrun = 0, 9
            b(testrun + 1) = 2
          END DO
          $OMP_PARALLEL FOR
          DO testrun = 0, 9
            c(testrun + 1) = 2
          END DO
        $OMP_END PARALLEL
        $OMP_TARGET_EXIT_DATA copyout(a(0:9),b(0:9),c(0:9))

        DO testrun = 0, 9
          IF (abs(a(testrun + 1) - 1) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
          IF (abs(b(testrun + 1) - 1) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
          IF (abs(c(testrun + 1) - 1) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END